src/core/ashp.jl

```diff
@@ -0,0 +1,227 @@
+# REopt®, Copyright (c) Alliance for Sustainable Energy, LLC. See also https://github.com/NREL/REopt.jl/blob/master/LICENSE.
+
+struct ASHP <: AbstractThermalTech
+    min_kw::Real
+    max_kw::Real
+    installed_cost_per_kw::Real
+    om_cost_per_kw::Real
+    macrs_option_years::Int
+    macrs_bonus_fraction::Real
+    can_supply_steam_turbine::Bool
+    heating_cop::Array{Float64,1}
+    cooling_cop::Array{Float64,1}
+    heating_cf::Array{Float64,1}
+    cooling_cf::Array{Float64,1}
+    can_serve_dhw::Bool
+    can_serve_space_heating::Bool
+    can_serve_process_heat::Bool
+    can_serve_cooling::Bool
+    force_into_system::Bool
+    back_up_temp_threshold::Real
+end
+
+
+"""
+ASHP_SpaceHeater
+
+If a user provides the `ASHP_SpaceHeater` key then the optimal scenario has the option to purchase 
+this new `ASHP` to meet the heating load in addition to using the `ExistingBoiler`
+to meet the heating load. 
+
+```julia
+function ASHP_SpaceHeater(;
+    min_ton::Real = 0.0, # Minimum thermal power size
+    max_ton::Real = BIG_NUMBER, # Maximum thermal power size
+    installed_cost_per_ton::Union{Real, nothing} = nothing, # Thermal power-based cost
+    om_cost_per_ton::Union{Real, nothing} = nothing, # Thermal power-based fixed O&M cost
```

Comment:
@zolanaj @atpham88 For configs 1 and 3, where we keep the existing boiler and/or chiller, we need to use the O&M number/default in the requirement doc. Configs 2 and 4 have a default of 0/ton/year because we assume ASHP has an equivalent O&M as the existing HVAC which is now retired.

Reply:
I set the default o&m to 40 and set o&m to 0 when force into system here https://github.com/NREL/REopt.jl/pull/372/commits/7297d80518c17dcab07ecea9b06682d9ae116406